### PR TITLE
Funcotator - Adding cDNA Strings for Intronic Variants

### LINKED ISSUE #5187: Funcotator should create a cDNA annotation for Intronic Splice Sites
### Tool(s) or class(es) involved
_Funcotator,GencodeFuncotationFactory_

### Description
When annotating a variant and the `VariantClassification` is `SPLICE_SITE, INTRON`, Funcotator should include a non-empty `cDNA` annotation.

This `cDNA` annotation should include the number of bases that the start of the variant is away from the exon, as well as the reference and alternate alleles of the variant in question.

Concretely:
    `c.e[EXON NUMBER][+|-][BASES FROM EXON][REF ALLELE]>[ALT ALLELE]`

For example:
    c.e2-1A>G
    Where:
        2 = the number of the closest exon (1-based)
        -1 = number of bases away from the exon (1 before)
        A = Reference allele
        G = Alternate allele

This feature is requested for the clinical pipeline.

### PR DESCRIPTION
**Pull request by jonn-smith**
Added in cDNA strings for intronic variants.
    
    Now intronic variants produce a cDNA string.  This string is created in
    a similar way to how the Codon Change String is created for intronic
    variants:
    
    If the transcript containing the variant has no exons, the cDNA string
    will be empty.
    
    If the transcript in which the variant occurs contains at least 1 exon, the string will be non-empty and of
    the form:
        c.e[EXON NUMBER][+|-][BASES FROM EXON][REF ALLELE]>[ALT ALLELE]
    Concretely:
        c.e2-1A>G
    Where:
         2 = the number of the exon to which the given variant start is closest
        -1 = number of bases away from the exon (1 before)
         A = Reference allele
         G = Alternate allele
    
    Fixes #5187

### CODE REVIEW

File: src/main/java/org/broadinstitute/hellbender/tools/funcotator/FuncotatorUtils.java [hunk 1]
```diff
@@ -1649,6 +1649,86 @@ public static String convertHG19ContigToB37Contig( final String hg19Contig ) {
...
+     *     -1 = number of bases away from the exon (1 before)
+     *      A = Reference allele
+     *      G = Alternate allele
+     * @param variantStart The start position (1-based, inclusive) in genomic coordinates of the variant.
+     * @param exonList The {@link List<Locatable>} representing the exons in the transcript in which this variant occurs.
+     * @param strandCorrectedRefAllele A {@link String} containing the bases of the reference allele, which are correct for strandedness (i.e. if on a transcript on the {@link Strand#NEGATIVE} strand, the string has already been reverse-complemented).
+     * @param strandCorrectedAltAllele A {@link String} containing the bases of the alternate allele, which are correct for strandedness (i.e. if on a transcript on the {@link Strand#NEGATIVE} strand, the string has already been reverse-complemented).
+     * @return A {@link String} representing the cDNA change for the given data.  Will be empty if the given {@code exonList} is empty.
+     */
+    public static String createIntronicCDnaString(final int variantStart,
```
Comment (LeeTL1220) [inline src/main/java/org/broadinstitute/hellbender/tools/funcotator/FuncotatorUtils.java]: Can  all of these parameters be `null`?  If not, please add a comment in the parameter javadoc (`Never {@code null}` and check the values with `Utils.nonNull(...)`  ...  This should be done for all public methods.

File: src/main/java/org/broadinstitute/hellbender/tools/funcotator/FuncotatorUtils.java [hunk 2]
```diff
@@ -1649,6 +1649,86 @@ public static String convertHG19ContigToB37Contig( final String hg19Contig ) {
...
+    }
+
+
+    /**
+     * Get the index of the exon that is closest to the given start position of a variant.
+     * Checks both before and after the start position to get the closest exon such that it may occur before the
+     * variant, within the variant, or after the variant.
+     * If there are no exons in the transcript, will return {@code null}.
+     * @param variantStartPos Start position (1-based, inclusive) in genomic coordinates of a variant.
+     * @param exonList The {@link List<Locatable>} representing the exons in the transcript in which this variant occurs.
```
Comment (LeeTL1220) [inline src/main/java/org/broadinstitute/hellbender/tools/funcotator/FuncotatorUtils.java]: "Never {@code null}"

**Comment by LeeTL1220** [comment]
@jonn-smith Make sure to check that travis errors are transient.

File: src/main/java/org/broadinstitute/hellbender/tools/funcotator/FuncotatorUtils.java (hunk 1 above)
Comment (jonn-smith) [reply src/main/java/org/broadinstitute/hellbender/tools/funcotator/FuncotatorUtils.java]: It was an oversight leaving it out.

Fixed!

File: src/main/java/org/broadinstitute/hellbender/tools/funcotator/FuncotatorUtils.java (hunk 2 above)
Comment (jonn-smith) [reply src/main/java/org/broadinstitute/hellbender/tools/funcotator/FuncotatorUtils.java]: Fixed!

**Pull request merged**